arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -1056,44 +1056,48 @@ Result RocksDBCollection::lookupKey(
     transaction::Methods* trx, std::string_view key,
     std::pair<LocalDocumentId, RevisionId>& result,
     ReadOwnWrites readOwnWrites) const {
+  return doLookupKey(trx, key, result, readOwnWrites, false);
+}
+
+Result RocksDBCollection::lookupKeyForUpdate(
+    transaction::Methods* trx, std::string_view key,
+    std::pair<LocalDocumentId, RevisionId>& result) const {
+  return doLookupKey(trx, key, result, ReadOwnWrites::yes, true);
```

Comment:
The `ReadOwnWrites::yes` here may be overkill for single document operations. In these, I don't think it is necessary to observe our own writes (because it is a single document transaction). But I may be wrong.

Reply:
That is true for single document operations, but since this is generic code we do not have the relevant information. Besides, for single document operations we have a specialized RocksDBMethods implementation, so it doesn't really matter performance wise.